shared_model/backend/protobuf/block.hpp

```diff
@@ -80,6 +80,11 @@ namespace shared_model {
         return true;
       }
 
+      bool clearSignatures() override {
+        signatures_->clear();
+        return (signatures_->size() == 0);
```

Comment:
Is there any case when it can return false?
P.s: If you're considering multi-thread behavior, e.g: here goes cleaning and somewhere else Lazy field invalidated, then there's still an rc-bug may happen. Moreover I don't think there's an "easy" solution of that problem with the current design

Reply:
move to Signable class

Reply:
Comment may be useful then, it is not obvious